### PR TITLE
[MM-34342] Migrate 'components/searchable_user_list' to TypeScript

### PR DESCRIPTION
**Pull request by sinansonmez**
#### Summary
This PR converts two files under 'components/searchable_user_list' folder to TypeScript

#### Ticket Link  
Fixes https://github.com/mattermost/mattermost-server/issues/17281
Jira ticket: https://mattermost.atlassian.net/browse/MM-34342
#### Screenshots
not applicable

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by sinansonmez** [comment]
Even though this is just a migration to TypeScript, there was also change in `searchable_user_list` component (i.e. `UserList` component is wrapped with `Scrollbars` to move from string ref). I kindly ask QA review on this to make sure that there is no change in behavior. 

I already tested this manually on the following places
- Team members
- Channel members
- System members (in system console)

File: components/searchable_user_list/searchable_user_list_container.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    };
+    focusOnMount?: boolean;
+}
+
+type State = {
+    term: string;
+    page: number;
+};
+
+export default class SearchableUserListContainer extends React.PureComponent<Props, State> {
```
Comment (BenCookie95) [inline components/searchable_user_list/searchable_user_list_container.tsx:46]: I wonder if we could turn this into a functional component? We are trying to use functional components for smaller components

**Review by BenCookie95 — CHANGES_REQUESTED** [review]
Thanks for the pull request! A few small comments

File: components/searchable_user_list/searchable_user_list.tsx [hunk 2]
```diff
@@ -14,97 +17,157 @@ import {t} from 'utils/i18n';
...
+    filter?: string;
+    renderFilterRow?: (handleFilter: ((event: React.FormEvent<HTMLInputElement>) => void) | undefined) => JSX.Element;
+    page: number;
+    term: string;
+    onTermChange: (term: string) => void;
+    intl: IntlShape;
+    isDisabled?: boolean;
+
+    // the type of user list row to render
+    rowComponentType?: unknown;
```
Comment (BenCookie95) [inline components/searchable_user_list/searchable_user_list.tsx:58]: I wonder if this can be `JSX.Element`?

Comment (sinansonmez) [reply components/searchable_user_list/searchable_user_list.tsx:58]: `JSX.Element` didn't work. 

The type of the passed element is `NamedExoticComponent<JSX.LibraryManagedAttributes<UserListRowWithError, (Omit<ClassAttributes<InstanceType<UserListRowWithError>> & Props, "status"> & OwnProps) | (Omit<ClassAttributes<InstanceType<UserListRowWithError>> & Readonly<...>, "status"> & OwnProps) | (Omit<ClassAttributes<InstanceType<UserListRowWithError>> & Readonly<...> & Readonly<{children?: ...}>, "status"> & OwnProps)>> & hoistNonReactStatics.NonReactStatics<UserListRowWithError> & {WrappedComponent: UserListRowWithError}`

`React.ComponentType<unknown>` is accepted, still not the the best but better than `unknown`

File: components/searchable_user_list/searchable_user_list_container.tsx (hunk 1 above)
Comment (sinansonmez) [reply components/searchable_user_list/searchable_user_list_container.tsx:46]: Done. Good to know, thanks. I will keep in my in future PRs

**Review by M-ZubairAhmed — APPROVED** [review]
this look good to me too.

**Comment by M-ZubairAhmed** [comment]
> Even though this is just a migration to TypeScript, there was also change in `searchable_user_list` component (i.e. `UserList` component is wrapped with `Scrollbars` to remove string ref). I kindly ask QA review on this to make sure that there is no change in behavior.
> 
> I already tested this manually on the following places
> 
>     * Team members
> 
>     * Channel members
> 
>     * System members (under system console)

Thanks good to know when doing QA :)

**Review by BenCookie95 — APPROVED** [review]

**Comment by metanerd** [comment]
/e2e-test
Sorry for spam. Should work now :crossed_fingers:

**Comment by jgilliam17** [comment]
Thank you @sinansonmez 
Tested, looks good to merge.
- Verified searchable user lists (team, channels, users etc) - working as before.
- [E2E report](https://mattermost-cypress-report.s3.amazonaws.com/169457-6d99e07-onprem-ent-webapp-pr-9924-mattermost/mm-ee-test6d99e07/mochawesome.html) looks good, no PR related failures.

**Pull request merged**